serverlessworkflow/modules/ROOT/pages/cloud/enhanced-service-discovery.adoc

```diff
@@ -0,0 +1,128 @@
+
+
+The Enhanced Service Discovery allows you to have a static URI that will define the Kubernetes resource that you use
+to perform http requests, the following sketch demonstrate how the SWUrl is composed:
+
+[source,shell]
+----
+kubernetes:<group>/<version>/<kind>/<namespace>/<name>?<attributeName>=<attributeValue>
+\________/ \_____/ \_______/ \____/ \_________/ \____/ \______________________________/
+  scheme    group   version   kind   namespace   name   additional resource attributes
+            \____________________/                       \__________________________/
+                     GVK                                  Supported value: port_name
+----
+
+On the scheme, the following values are supported:
+
+* **kubernetes**
+* **openshift**
+* **knative**
+
+As GVK, for now, the following Kubernetes Resources are supported:
+
+* **v1/service**
+* **serving.knative.dev/v1/service**
+* **v1/pod**
+* **apps/v1/deployment**
+* **apps.openshift.io/v1/deploymentconfig**
+* **apps/v1/statefulset**
+* **route.openshift.io/v1/route**
+* **networking.k8s.io/v1/ingress**
+
+[TIP]
+====
+Note that the _GVK_(Group, Version and Kind) and the _Resource Name_ are mandatory and can't be empty, otherwhise, the engine will throw a validation issue and the discovery will not be performed. The **namespace** is not mandatory, however, if empty, the current namespace or context will be used.
+====
+
+== Configuration
+
+There are two **build time** system properties, which are:
+
+* `quarkus.kogito.sw.discovery.enabled`: Disable the service discovery. Defaults to _true_.
+* `quarkus.kogito.kubernetes-client.okhttp.static.logger`: Weather or not to print the `http`communication between the client and the Kubernetes API. Defaults to _false_.
+**  Note that, this system property can change in the future since this feature will be ported to the _quarkus-kubernetes-client_ extension. For more info please refer to this link:{quarkus_issue_url}[link].
+
+
+== Startup time: discovery enabled x disabled
+
+[tabs]
+====
+Service Discovery Enabled::
++
+[source,shell]
+----
+(main) sw-service-discovery 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 2.360s. Listening on: http://0.0.0.0:8080
+----
+Service Discovery Disabled::
++
+[source,shell]
+----
+(main) sw-service-discovery 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.507s. Listening on: http://0.0.0.0:8080
+----
+====
+
+[IMPORTANT]
+====
+When using this feature, you need to balance if the delayed startup time is something
+that your application can afford.
+====
+
+== How does it work?
+
+The service discovery is performed at the Quarkus _STATIC_INIT_ time, which means, before the application gets started. It scans the system properties and searches for the _SWUrl_ pattern, if at least one is found, the engine will parse the SWUrl and then query the Kubernetes API for the given resource and override the given system property. Let's see a small example of it:
```

Comment:
system property terminology might be confusing. 
In Java, a system property is a property passed to the JVM, usually through command line, using -D prefix. 
What you are actually doing in the code is scanning Quarkus configuration, of which Java system properties are just a subset. 
Hence I think the term quarkus configuration (and a link to Quarkus configuration guide) will be better in that context